nrpy/infrastructures/BHaH/seobnr/SEOBNR_waveform_C_codegen_library.py

```diff
@@ -272,3 +272,169 @@ def register_CFunction_SEOBNRv5_NQC_corrections() -> Union[None, pcg.NRPyEnv_typ
         body=body,
     )
     return cast(pcg.NRPyEnv_type, pcg.NRPyEnv())
+
+
+def register_CFunction_SEOBNRv5_aligned_spin_IMR_waveform() -> (
+    Union[None, pcg.NRPyEnv_type]
+):
+    """
+    Register CFunction for evaluating the (2,2) IMR mode for the SEOBNRv5 waveform.
+
+    :return: None if in registration phase, else the updated NRPy environment.
+    """
+    if pcg.pcg_registration_phase():
+        pcg.register_func_call(f"{__name__}.{cast(FT, cfr()).f_code.co_name}", locals())
+        return None
+
+    includes = ["BHaH_defines.h", "BHaH_function_prototypes.h"]
+    desc = """Evaluate SEOBNRv5 (2,2) mode."""
+    cfunc_type = "int"
+    name = "SEOBNRv5_aligned_spin_IMR_waveform"
+    params = "commondata_struct *restrict commondata"
+    bob_wf = BOB.BOB_aligned_spin_waveform_quantities()
+    body = """
+int i;
+REAL *restrict times_old = (REAL *)calloc(commondata->nsteps_inspiral,sizeof(REAL));
```

Comment:
It's generally not a good idea to use calloc when malloc will do, as initializing to zero might hide uninitialized memory bugs or issues that could otherwise be caught during debugging, and it may add unnecessary overhead if zero-initialization is not required.